convert_tendl_rand.py

```diff
@@ -0,0 +1,360 @@
+        'fileSize' : 2000,
+        'isItENDF' : True,
+        'gunzip'   : False
+    }, 
+    'Fe58' : {
+        'release'  : 'ENDF2017',
+        'filename' : 'Fe058',
+        'webname'  : 'Fe058',
+        'fileNum'  : 501,
+        'downSize' : 280,
+        'fileSize' : 1100,
+        'isItENDF' : True,
+        'gunzip'   : False
+    }
+}
+
+downloadFileSize = 0
+uncompressedFileSize = 0
+NumOfFiles = 0
+
+for i in list_:
+    downloadFileSize += nuclide_details[i]['downSize']
+    uncompressedFileSize += nuclide_details[i]['fileSize']
+    NumOfFiles += nuclide_details[i]['fileNum']
+
+downloadSize = '{} MB'.format(downloadFileSize)
+uncomFileSize = '{} MB'.format(uncompressedFileSize)
+if (downloadFileSize > 1000): downloadSize = '{} GB'.format(downloadFileSize/1000)
+if (uncompressedFileSize > 1000): uncomFileSize = '{} GB'.format(uncompressedFileSize/1000)
+
+    
+
+download_warning = """
+WARNING: This script will download {} of 
+data, which is {} of data when processed. 
+This corresponds to {} random crossections.
+
+The nuclides to be processed are: 
+{}
+
+Are you sure you want to continue? ([y]/n)
+""".format(downloadSize, uncomFileSize,
+          NumOfFiles, list_)
+
+
+response = input(download_warning) if not args.batch else 'y'
+if response.lower().startswith('n'):
+    sys.exit()
+
+# ==============================================================================
+# DOWNLOAD FILES FROM WEBSITE
+
+files_complete = []
+for nucs in list_:
+    # Establish connection to URL
+    url = release_details[nuclide_details[nucs]['release']]['base_url'] + nuclide_details[nucs]['webname'] + release_details[nuclide_details[nucs]['release']]['ending']
+    print("Downloading {}...".format(nucs))
+    downloaded_file = download(url)
+    #print(url)
+    #files_complete.append(downloaded_file)
+
+# ==============================================================================
+# EXTRACT FILES FROM TGZ
+
+for nucs in list_:
+    f = nuclide_details[nucs]['webname'] + '.random.tgz'
+    suffix = nucs
+    isItENDF = nuclide_details[nucs]['isItENDF']
+    if isItENDF:
+        outDir = endf_files_dir
+    else:
+        outDir = ace_files_dir
+    
+    with tarfile.open(f, 'r') as tgz:
+        print('Extracting {0}...'.format(f))
+        tgz.extractall(path=os.path.join(outDir, suffix))
+        
+# ==============================================================================
+# Format file names 
+
+for nucs in list_:
+    f = nuclide_details[nucs]['webname'] + '.random.tgz'
+    isItENDF = nuclide_details[nucs]['isItENDF']
+    numFiles = nuclide_details[nucs]['fileNum']
+    if isItENDF:
+        outDir = os.path.join(endf_files_dir, nucs)
+        prefix = 'n-'
+        suffix = '-rand-'
+        for i in range(0,numFiles):
+            if i < 10:
+                OldNumber = '000' + str(i)
+            elif i < 100:
+                OldNumber = '00' + str(i)
+            elif i < 1000:
+                OldNumber = '0' + str(i)
```

Comment:
`f'{i:04}'` would give a zero-padded string of length 4, achieving the same effect with one line of code